jetty-util/src/main/java/org/eclipse/jetty/util/thread/strategy/AdaptiveExecutionStrategy.java

```diff
@@ -134,19 +135,21 @@ public class AdaptiveExecutionStrategy extends ContainerLifeCycle implements Exe
     private final LongAdder _epcMode = new LongAdder();
     private final Producer _producer;
     private final Executor _executor;
+    private final boolean _useVirtualThreads;
     private final TryExecutor _tryExecutor;
     private final Runnable _runPendingProducer = () -> tryProduce(true);
     private State _state = State.IDLE;
     private boolean _pending;
 
     /**
-     * @param producer The produce of tasks to be consumed.
+     * @param producer The producer of tasks to be consumed.
      * @param executor The executor to be used for executing producers or consumers, depending on the sub-strategy.
      */
     public AdaptiveExecutionStrategy(Producer producer, Executor executor)
     {
         _producer = producer;
         _executor = executor;
+        _useVirtualThreads = VirtualThreads.isUseVirtualThreads(executor);
```

Comment:
Is grabbing this at construction too early?
I know it is nice to have as a final, but I think it is probably more correct to look at the value in `doStart` as it might be set on the executor after the strategy has been built (but not started).

Reply:
Good point, done.